src/controller/java/AndroidDeviceControllerWrapper.cpp

```diff
@@ -184,13 +184,16 @@ AndroidDeviceControllerWrapper * AndroidDeviceControllerWrapper::AllocateNew(
         chip::Credentials::SetDeviceAttestationVerifier(GetDefaultDACVerifier(testingRootStore));
     }
 
-    *errInfoOnFailure = getICDClientStorage()->Init(wrapperStorage, &wrapper->mSessionKeystore);
-    if (*errInfoOnFailure != CHIP_NO_ERROR)
+    auto icdStorage = getICDClientStorage();
+    if (icdStorage != nullptr && !(icdStorage->IsInitialized()))
```

Comment:
OK, but what if the wrapper's storage and keystore don't match the existing ones on the icdStorage?  How is this supposed to work?

Reply:
we don't support changing wrapper's storage and keystore, right?

Reply:
> we don't support changing wrapper's storage and keystore, right?

I don't know; the wrapper is part of the Android API and I am not familiar with it.  

But if there are two calls to `AndroidDeviceControllerWrapper::AllocateNew`, won't those have different `wrapperStorage` and `&wrapper->mSessionKeystore` values?  Is calling `AllocateNew` twice just not supported?  But this whole issue that this PR is claiming to fix seems to be about "across multiple controoler
 [sic]", which means multiple calls to `AndroidDeviceControllerWrapper::AllocateNew`, I would think.

Let me put it this way: at this point, short of an actual test exercising this code in the scenario of https://github.com/project-chip/connectedhomeip/issues/36346, I don't believe this fixes https://github.com/project-chip/connectedhomeip/issues/36346 at all.

Reply:
Just update code to shutdown icd client storage when destroying android controller, thanks